model/FullText.inc.php

```diff
@@ -52,58 +53,33 @@ public static function indexItem(Zotero_Item $item, $data) {
 			. implode(', ', array_fill(0, sizeOf($params), '?')) . ")";
 		Zotero_DB::query($sql, $params, Zotero_Shards::getByLibraryID($libraryID));
 		
-		// Add to Elasticsearch
-		self::indexItemInElasticsearch($libraryID, $key, $version, $timestamp, $data);
-		
-		Zotero_DB::commit();
-	}
-	
-	private static function indexItemInElasticsearch($libraryID, $key, $version, $timestamp, $data) {
-		$type = self::getWriteType();
-		
-		$id = $libraryID . "/" . $key;
-		$doc = [
-			'id' => $id,
+		// Add to S3
+		$json = [
 			'libraryID' => $libraryID,
-			'content' => (string) $data->content,
-			// We don't seem to be able to search on _version, so we duplicate it here
+			'key' => $key,
 			'version' => $version,
-			// Add "T" between date and time for Elasticsearch
+			'content' => (string)$data->content,
 			'timestamp' => str_replace(" ", "T", $timestamp)
 		];
+		
 		foreach (self::$metadata as $prop) {
 			if (isset($data->$prop)) {
-				$doc[$prop] = (int) $data->$prop;
+				$json[$prop] = (int)$data->$prop;
 			}
 		}
+		
+		$json = json_encode($json, JSON_UNESCAPED_UNICODE | JSON_UNESCAPED_SLASHES);
+		
+		$s3Client = Z_Core::$AWS->createS3();
 		$start = microtime(true);
-		$doc = new \Elastica\Document($id, $doc, self::$elasticsearchType);
-		$doc->setVersion($version);
-		$doc->setVersionType('external');
-		try {
-			$response = $type->addDocument($doc);
-		}
-		catch (Exception $e) {
-			$msg = $e->getMessage();
-			if (preg_match('/version conflict, current \[([0-9]+)\], provided \[([0-9]+)\]/', $msg, $matches)) {
-				if ($matches[1] == $matches[2]) {
-					error_log("WARNING: " . $msg);
-					return;
-				}
-			}
-			throw $e;
-		}
-		StatsD::timing("elasticsearch.client.item_fulltext.add", (microtime(true) - $start) * 1000);
-		if ($response->hasError()) {
-			$msg = $response->getError();
-			if (preg_match('/version conflict, current \[([0-9]+)\], provided \[([0-9]+)\]/', $msg, $matches)) {
-				if ($matches[1] == $matches[2]) {
-					error_log("WARNING: " . $msg);
-					return;
-				}
-			}
-			throw new Exception($response->getError());
-		}
+		$s3Client->putObject([
+			'Bucket' => Z_CONFIG::$S3_BUCKET_FULLTEXT,
+			'Key' => $libraryID . "/" . $key,
+			'Body' => $json
```

Comment:
It seems that versioning can't be disabled once enabled. Do we really need it?

> Once you version-enable a bucket, it can never return to an unversioned state. You can, however, suspend versioning on that bucket.